projects/fw_103_hw/src/main.c

```diff
@@ -0,0 +1,51 @@
+#include "adc.h"
+#include "delay.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+
+static void prv_button_interrupt_handler(const GpioAddress *address, void *context) {
+  uint16_t A6_data = 0;
+  AdcChannel *channel = (AdcChannel *)context adc_read_raw(channel, &A6_data);
+  LOG_DEBUG("Pin A6 data: %d\n", A6_data);
+}
+
+int main(void) {
+  interrupt_init();
+  gpio_init();
+  gpio_it_init();
+
+  GpioAdress pin_address = { .port = GPIO_PORT_A, .pin = 6 };
+
+  GpioAdress button_address = { .port = GPIO_PORT_B, .pin = 2 };
+
+  GpioSettings settings = {
+    GPIO_DIR_IN,
+    GPIO_STATE_LOW,
+    GPIO_RES_NONE,
+    GPIO_ALTFN_ANALOG,
+  };
```

Comment:
Please use the GNU-style struct initialization syntax: `.direction = GPIO_DIR_IN`, `.state = GPIO_STATE_LOW`, etc. It makes it easier to read and lets the struct's semantics not depend on the order of the items in the initializer.

Also, surely there's a better name for the gpio settings than a generic `settings` when you've got other settings structs in the program.